internal/plan/plan.go

```diff
@@ -622,6 +695,33 @@ func (p *Plan) StopOrder(names []string) ([]string, error) {
 	return order(p.Services, names, true)
 }
 
+// SetServiceArgs sets the service arguments provided by "pebble run --args"
+// to their respective services. Additionally, for a service the arguments
+// are also updated in the layer of the highest order that defines the service
+// so that the arguments to a service can persist if it is not affected
+// by a "pebble add" layer or similar.
+func (p *Plan) SetServiceArgs(serviceArgs map[string][]string) error {
```

Comment:
Yeah, this is the logic (or similar logic) that I think should be separated from the plan and go in the service manager.